src/sync-external-store--stateful-store/store-item.ts

```diff
@@ -0,0 +1,9 @@
+import { statefulStates } from "../shared/index.js";
+
+export type StoreState = Record<string, StoreItem<unknown>>;
+
+export type StoreItem<TData> = {
+  data: TData;
+  error: null | string;
+  state: typeof statefulStates;
```

Comment:
This doesn't seem right... I don't think your typescript types are flowing through correctly. I need to pull this down...